apps/internal/base/base.go

```diff
@@ -380,6 +393,12 @@ func (b Client) AcquireTokenOnBehalfOf(ctx context.Context, onBehalfOfParams Acq
 		return ar, err
 	}
 	authParams, err := b.AuthParams.WithTenant(onBehalfOfParams.TenantID)
+	//if instance discovery is disabled, add authority to known hosts, so that Alias Lookup is not done
+	//withTenant seems to reset the previous AuthParams, so have to add this knownAuthorityHosts field again
+	if authParams.AuthorityInfo.InstanceDiscoveryDisabled {
+		authParams.KnownAuthorityHosts = append(authParams.KnownAuthorityHosts, authParams.AuthorityInfo.Host)
```

Comment:
If this is necessary, it conceals a bug. `b.AuthParams.WithTenant(...)` returns a copy of `b.AuthParams` having the specified tenant; it doesn't modify `KnownAuthorityHosts`. If the copy has an empty or otherwise unexpected value for `KnownAuthorityHosts`, I expect `b.AuthParams` to have the same unexpected value.

Reply:
What you said is correct. I think just InstanceDiscoveryDisabled was being defaulted to false. Removing this line ran the tests fine.